GitCommands/Git/Executable.cs

```diff
@@ -230,12 +231,15 @@ public StreamReader StandardError
             public void WaitForInputIdle() => _process.WaitForInputIdle();
 
             /// <inheritdoc />
-            public Task<int> WaitForExitAsync() => _exitTaskCompletionSource.Task;
+            public Task<int> WaitForExitTaskAsync() => _exitTaskCompletionSource.Task;
 
             /// <inheritdoc />
-            public int WaitForExit()
+            public Task WaitForExitAsync(CancellationToken token) => _process.WaitForExitAsync(token);
+
+            /// <inheritdoc />
+            public int WaitForExitTask()
```

Comment:
This name is confusing and wrong. It returns an `int` and not a `task`. The same goes for the async version.

Reply:
The method "waits for the exitTask" created when the process exits, including the extra work to add to log etc.
It is not a normal `WaitForExit()` that waits for the process task to exit.

I still think the name is better.
But what about `WaitForProcessExit()`?

Reply:
It is the non-async form of `WaitForExitAsync` so should be called `WaitForExit` in order to be consistent with the rest of the framework (e.g. `Read`/`ReadAsync`, `Write`/`WriteAsycn`, ...).

Reply:
But there are two variants of WaitForExit() now: One that waits for the process to exit (that I added, cancellable) and the existing that waits for the additional exitTask to exit. I find the new method to follow the intention of the WaitForExit() method better, but I can rename that instead.